### PR TITLE
Revamp documentation to be better suitable for newcommers.

### PR DESCRIPTION
**Pull request by SijmenHuizenga**
Over the past months the docs have been growing organically. There is so much useful information in there! Recently we have been getting feedback that it isn't very clear how to start. It is time to re-evaluate the documentation structure. 

In the original structure we tried to split documentation for users and developers. Looking back, I think these groups are not so different after all. A developer wants to run the simulator and ros bridge just like a user would, only difference being that the developer will more likely choose to run the unreal engine project from the Unreal Editor.

In this pr I combined the `how-to-simulate`, `how-to-develop`, `get-ready-to-develop` and `building-ros.md` into a single `getting started` guide. This guide walks you through the steps to get the project running. It offers choices and explains any pros/cons between the options. This solves #118 because now there is a better description of how the system can be used.

I moved all software installation instructions into software-install-instructions.md. This file is referenced whenever the reader needs to have some piece of software installed.

Most of the changes in this pr are moving of text. However, I did add more instructions where I felt this was required.

To the reviewers: would you like to read the `docs/getting-started.md` from top to bottom and check if it has a logical structure and flow?

Feel free to commit spelling errors directly to this branch.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+* 30GB free SSD storage (120GB when building the unreal project from source)
+* Recent NVidia card with Vulkan support and 3 GB of memory. (You can check the video card drivers by running `vulkaninfo`)
+
+If your computer does not suffice you can use a remote workstation on Google Cloud Platform.
+Read [this tutorial](gcp-remote-workstation.md) on how to setup your virtual workstation.
+
+The simulator will load settings from the file `Formula-Student-Driverless-Simulator/settings.json` in your **home directory**.
+This file contains the sensor configuration of the car.
+Copy-paste the contents of the [settings.json file at the root of this repository](https://github.com/FS-Online/Formula-Student-Driverless-Simulator/blob/master/settings.json) inside.
+This should get you started with the default sensor configuration, feel free to try your own custom sensor suite.
```
Comment (davidoort) [inline docs/getting-started.md:25]: This part confuses me. Currently I'm running on an Ubuntu instance and I just have a settings.json file inside the clone of this repository, so clearly these steps do not apply in all cases. We should make clear if the settings.json file has to be in the home directory or the Formula-Student-Driverless-Simulator folder. Adding something like, unless you have cloned this repository in the home directory, copy-paste the contents...

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+### From release binaries
+
+The simulator is distributed as binaries on every release.
+[At this moment only windows binaries are released. For now, if you are on Ubuntu you will have to run the simulator from the Unreal Engine Editor.](https://github.com/FS-Online/Formula-Student-Driverless-Simulator/issues/107)
+We reccommend Windows for competitoins because it offers a bit better performance and stability.
```
Comment (davidoort) [inline docs/getting-started.md:31]: Do you mean to say that the simulator will be running on Windows during the competition?

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+If you havn't setup your ssh keys, you can clone using https by running the following command:
+```
+git clone https://github.com/FS-Online/Formula-Student-Driverless-Simulator.git --recurse-submodules
+```
+
+**THE REPO HAS TO BE CLONED IN THE HOME DIRECTORY!**. So the repo location should be `$HOME/Formula-Student-Driverless-Simulator`.
+Why you ask? Because we couldn't get relative paths in the C++ code to work so now we have hard-coded some paths to the home directory.
+I know yes it is ugly but it works. If you are bothered by it I would welcome you to open a pr with a fix.
+
+If this folder already exists as a result of any previous step, you can get your settings.json out, delete the folder (or rename it to something like UEProject or UESim) and after cloning the repo place the settings.json file back (this time inside the root of the repository where the default settings.json lives).
```
Comment (davidoort) [inline docs/getting-started.md:126]: This is hard to follow, could you rewrite it please?

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+If you are on Windows and cloned this repository in a Windows directory, go into the cloned repo and run `git config core.fileMode false` to ignore file mode changes. 
+If you want to share the the clone directory with the Ubuntu WSL system, create a symlink from `~/Formula-Student-Driverless-Simulator` to `~/Formula-Student-Driverless-Simulator`
```
Comment (davidoort) [inline docs/getting-started.md:129]: do you mean from ~/Formula-Student-Driverless-Simulator to /mnt/c/Users/developer/Formula-Student-Driverless-Simulator?

Comment (SijmenHuizenga) [reply docs/getting-started.md]: This is confusing indeed. Changed it now.

File: docs/getting-started.md (hunk 3 above)
Comment (SijmenHuizenga) [reply docs/getting-started.md]: Yes!

File: docs/getting-started.md (hunk 2 above)
Comment (SijmenHuizenga) [reply docs/getting-started.md]: Yes. Rewrote sentence.

File: docs/getting-started.md (hunk 1 above)
Comment (SijmenHuizenga) [reply docs/getting-started.md]: Clarified it, how do you feel about it now?

Comment (davidoort) [reply docs/getting-started.md]: great!
